### PR TITLE
[ATO-1151] Parentheses in button title overwrite button payload in rasa shell

### PR DESCRIPTION
**Pull request by Urkem**
**Proposed changes**:
- Fix [ATO-1151] 

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

[ATO-1151]: https://rasahq.atlassian.net/browse/ATO-1151?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: changelog/12534.bugfix.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Buttons in `rasa shell` are now correctly parsed.
```
Comment (ancalita) [inline changelog/12534.bugfix.md:1]:
```suggestion
Rich responses containing buttons with parentheses characters are now correctly parsed.
```
Added a bit more detail, and also I don't think this is just a rasa shell utility, please double-check and reintroduce if it's just rasa shell.

File: changelog/12534.bugfix.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+Previously any characters between `()` in response took precedence.
```
Comment (ancalita) [inline changelog/12534.bugfix.md:2]:
```suggestion
Previously any characters found between the first identified pair of `()` in response button took precedence.
```

File: tests/cli/test_utils.py [hunk 3]
```diff
@@ -355,3 +356,18 @@ def test_validate_assistant_id_in_config_preserves_comment() -> None:
...
     # reset input files to original state
     rasa.shared.utils.io.write_yaml(original_config_data, config_file, True)
+
+
+@pytest.mark.parametrize(
+    "text_input, button",
+    [
+        ("hi this is test text\n", "hi this is test text"),
+        ("hi this is test text (/button_one)", "/button_one"),
+        ("hi this is test text (and something) (/button_one)", "/button_one"),
```
Comment (ancalita) [inline tests/cli/test_utils.py:366]: Is this how the button gets represented from its yaml presentation? How does it end up in this string format? Just curious and don't have time to walk through the code 😅 
```
- title: "some text (blabla)"
   payload: /some_intent
```

**Review by ancalita — COMMENTED** [review]
Left two suggestions for improving the changelog, and asked a question for my understanding 😄

File: tests/cli/test_utils.py (hunk 3 above)
Comment (Urkem) [reply tests/cli/test_utils.py:366]: When someone selects the button in cli and presses enter it just adds the button after the title in text like `some text (blabla) (/some_intent)` but if it's free text it's `string + '\n'`, there is an example in the jira ticket.

File: changelog/12534.bugfix.md (hunk 1 above)
Comment (Urkem) [reply changelog/12534.bugfix.md:1]: Yup it's not just shell it's interactive too.

**Review by ancalita — APPROVED** [review]

File: tests/nlu/featurizers/test_lm_featurizer.py [hunk 4]
```diff
@@ -361,6 +361,7 @@ def evaluate_message_shapes(
             assert intent_sentence_vec is None
 
     @pytest.mark.timeout(120, func_only=True)
+    @pytest.mark.skip_on_windows
```
Comment (ancalita) [inline tests/nlu/featurizers/test_lm_featurizer.py:364]: Please add a Windows skip marker for `test_lm_featurizer_num_sub_tokens_process_training_data` as well.

**Pull request merged**